commands/kick.js

```diff
@@ -1,48 +1,55 @@
-const embed = require("../utils/embed.js");
+const { Permissions } = require('discord.js');
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const embed = require('../utils/embed.js');
 
 module.exports = {
-	name: "kick",
-	description: "Kicks a naughty user.",
-	aliases: ["boot", "begone"],
-	usage: "[@user]",
-	cooldown: 0,
-	guildOnly: true,
-	modOnly: true,
-	execute(message) {
-		const user = message.mentions.users.first();
+	data: new SlashCommandBuilder()
+		.setName('kick')
+		.setDescription('Kick a user.')
+		.addUserOption(option =>
+			option.setName('target')
+				.setDescription('The user to kick.')
+				.setRequired(true)
+		)
+		.addStringOption(option =>
+			option.setName('reason')
+				.setDescription('The reason for kicking this user.')
+		),
+	async execute(interaction) {
+		const member = interaction.options.getMember('target');
```

Comment:
Would it be worth having a helper method for fetching these 3 based on interaction as we use it regularly

Reply:
As discussed this is different in each function so probably not worth abstracting away.